### ISSUE TITLE
Double Update Accepted Event Due to Replay

### ISSUE THREAD

**Issue by mmcshane** [temporalio/sdk-go#1057]
## Expected Behavior
An update should be accepted (or rejected) exactly once

## Actual Behavior
If a workflow replays due to something like a worker crash while there is an update in-flight, we don't re-run the validator for the update but we _do_ call Accept() on the update state machine. That leads to an update.Acceptance message being sent on the WFTCompleted which is duly made durable into the history by the server.

Note that one might consider the fact that the server doesn't catch the mistake to _also_ be a bug. That will be covered with a separate issue. 

## Steps to Reproduce the Problem

There will be a test in temporalio/features for this (it's how the problem was found in the first place). The general flow is:

1. Start a workflow with separate activity and workflow workers
1. Start an update that itself launches an activity
1. Block in the activity
1. Stop or otherwise kill the workflow worker
1. Unblock the blocked activity
1. Start the workflow worker again
1. Observe that the workflow replays to rebuild state but that the incorrect extra UpdateAccepted event is added to the history. This history will not replay correctly.

## Specifications

  - Version: v1.21
  - Platform: all

**Comment by mmcshane** [comment]
Closed by #1059
